tools/run_tests/run_tests.py

```diff
@@ -850,12 +850,15 @@ def git(cmd, cwd=cwd):
     print 'copied to the docker environment.'
     time.sleep(5)
 
-  dockerfile_dirs = set([l.dockerfile_dir() for l in languages])
-  if len(dockerfile_dirs) > 1:
-    print 'Languages to be tested require running under different docker images.'
-    sys.exit(1)
-  dockerfile_dir = next(iter(dockerfile_dirs))
-
+  if 'gcov' in args.config:
```

Comment:
we should only use the multilang image when absolutely necessary
e.g.   -c gcov -l csharp  should still run under csharp image.

So i'd say let's only use the multilang image for 'gcov' and len(dockerfile_dirs) > 1.
Perhaps we could also print a message that the multilang image is getting used.